client/packages/common/src/ui/layout/tables/components/Cells/NumberInputCell/NumberCell.tsx

```diff
@@ -11,10 +11,18 @@ export const NumberCell = <T extends RecordWithId>({
   column,
   rowData,
   isError,
-}: CellProps<T>) => {
+  defaultValue,
+}: CellProps<T> & {
+  defaultValue?: string | number;
+}) => {
   const value = column.accessor({ rowData }) as number;
   const formattedValue = useFormatNumber().round(value, 2);
 
+  const displayValue =
+    (value === undefined || value === null) && !!defaultValue
```

Comment:
We need to explicitly check that `defaultValue !== undefined` here, as falsy values (0, "") return `false` for `!!`, and so setting a default of `""` displays as `0`. ;)

Reply:
Although with my earlier suggestion (default `defaultValue=""`), you won't need to check `defaultValue` at all here.